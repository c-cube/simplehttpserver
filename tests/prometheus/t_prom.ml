module P = Tiny_httpd_prometheus

let pf = Printf.printf
let reg = P.Registry.create ()
let c1 = P.Counter.create reg "t_c1"
let c2 = P.Counter.create reg "t_c2" ~descr:"more awesome than c1"
let g1 = P.Gauge.create reg ~tags:[ "level", "max" ] "yolo_gauge"

let h1 =
  P.Histogram.create reg ~descr:"latency"
    ~buckets:[ 0.01; 0.1; 0.5; 1.; 10. ]
    "latency"

let () =
  print_endline "==== first try ====";
  P.Counter.incr_by c1 42;
  P.Counter.incr c2;
  P.Gauge.set g1 2525;
  P.Histogram.add h1 0.2;
  P.Histogram.add h1 0.003;
  P.Histogram.add h1 0.002;
  P.Histogram.add h1 0.025;
  P.Histogram.add h1 0.9;
  P.Histogram.add h1 7.4;
  P.Histogram.add h1 22.2;
  P.Histogram.add h1 0.3;
  P.Histogram.add h1 0.4;
  P.Histogram.add h1 0.1;

  pf "```\n%s\n```\n" @@ P.Registry.emit_str reg

let () =
  print_endline "==== second try====";
  P.Counter.incr_by c1 11;
  P.Counter.incr c2;
  P.Gauge.set g1 42_000;
  P.Histogram.add h1 23.2;
  P.Histogram.add h1 0.2;

  pf "```\n%s\n```\n" @@ P.Registry.emit_str reg
