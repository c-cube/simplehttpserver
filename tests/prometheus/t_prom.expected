==== first try ====
```
# TYPE yolo_gauge gauge
yolo_gauge{level="max"} 2525
# HELP t_c2 more awesome than c1
# TYPE t_c2 counter
t_c2 1
# TYPE t_c1 counter
t_c1 42
# HELP latency latency
# TYPE latency histogram
latency{le="0.01"} 2
latency{le="0.1"} 4
latency{le="0.5"} 7
latency{le="1."} 8
latency{le="10."} 9
latency{le="+Inf"} 10
latency_count 10
latency_sum 31.530

```
==== second try====
```
# TYPE yolo_gauge gauge
yolo_gauge{level="max"} 42000
# HELP t_c2 more awesome than c1
# TYPE t_c2 counter
t_c2 2
# TYPE t_c1 counter
t_c1 53
# HELP latency latency
# TYPE latency histogram
latency{le="0.01"} 2
latency{le="0.1"} 4
latency{le="0.5"} 8
latency{le="1."} 9
latency{le="10."} 10
latency{le="+Inf"} 12
latency_count 12
latency_sum 54.930

```
